packages/glimmer-runtime/lib/builder.ts

```diff
@@ -329,7 +329,13 @@ class BlockTracker implements Tracker {
   }
 
   parentElement() {
-    return this.parent;
+    let parent = this.parent;
+    let firstNode = this.firstNode();
+    if (firstNode && parent && parent.children && parent.children.length === 0) {
```

Comment:
check for fragment should be `parent.nodeType === Node.DOCUMENT_FRAGMENT_NODE` and we should update this.parent to be the element but this really should be done when it is appended